lib-php/ReverseGeocode.php

```diff
@@ -85,6 +85,11 @@ protected function lookupInterpolation($sPointSQL, $fSearchDiam)
 
     protected function lookupLargeArea($sPointSQL, $iMaxRank)
     {
+        if(CONST_Search_WithinCountries
+            and $this->lookupInCountry($sPointSQL, $iMaxRank) == null){
+                return  null;
+        }
```

Comment:
This code is nicely readable but unfortunately leads to 1 - 2 additional unnecessary SQL queries to the database for every reverse query.

Split `lookupInCountry` in two functions, one which looks up the country code in the country grid (lines 107-120), one that looks up country information (lines 122ff). Then call here the first part, remembering the country code. And in line 102 call the second part using the remembered country code.

Reply:
Thank you so much. I added a commit that splits `lookupInCountry` in two functions and document the parameter.